mmf/datasets/databases/readers/feature_readers.py

```diff
@@ -178,6 +178,28 @@ def read(self, image_feat_path):
                     image_info.update(item["info"])
                 image_feature = item["feature"]
 
+        # Handle case of features with class probs
+        if (
+            image_info["features"].size == 1
+            and "features" in image_info["features"].item()
+        ):
```

Comment:
I thought about this. Made it an if statement because most of the functionality is identical. But might be better to seperate it from an understandability and it makes it less likely this will break other things. I can seperate it if you like.

Reply:
Yes, you can inherit that feature reader and update this specific part.